shared/desktop/app/app-state.js

```diff
@@ -115,45 +115,118 @@ export default class AppState {
-        applescript.execString(command, (err, result) => {
+      this.setDarwinLoginState()
+    } else if (isWindows) {
+      this.setWinLoginState()
+    }
+  }
+
+  setDarwinLoginState() {
+    const applescript = require('applescript')
+    this.checkMultiDarwinLoginItems()
+    try {
+      const appName = __DEV__ ? 'Electron Helper' : 'Keybase'
+      const command = this.state.openAtLogin
+        ? `tell application "System Events" to get the name of login item "${appName}"`
+        : `tell application "System Events" to delete login item "${appName}"`
+      applescript.execString(command, (err, result) => {
+        if (err) {
+          // eat the error here if changing from off to on
+          if (this.state.openAtLogin) {
+            applescript.execString(
+              `tell application "System Events" to make login item at end with properties {path:"${appBundlePath() ||
+                ''}", hidden:false, name:"${appName}"}`,
+              (err, result) => {
+                if (err) {
+                  console.log(`apple script error making login item: ${err}, ${result}`)
+                }
+              }
+            )
+          } else console.log(`apple script error: ${err}, ${result}`)
+        }
+      })
+    } catch (e) {
+      console.log('Error setting apple startup prefs: ', e)
+    }
+  }
+
+  // Remove all our entries but one to repair a previous bug. Can eventually be removed.
+  checkMultiDarwinLoginItems() {
+    const applescript = require('applescript')
+    const appName = __DEV__ ? 'Electron Helper' : 'Keybase'
+    try {
+      applescript.execString(
+        `tell application "System Events" to get the name of every login item`,
+        (err, result) => {
           if (err) {
-            console.log(`apple script error: ${err}, ${result}`)
+            console.log(`Error getting every login item: ${err}, ${result}`)
+          } else {
+            var foundApp = false
+            for (var loginItem in result) {
+              if (result[loginItem] === appName) {
+                if (!foundApp) {
+                  foundApp = true
+                  continue
+                }
+                console.log('login items: deleting ', appName)
+                applescript.execString(
+                  `tell application "System Events" to delete login item "${appName}"`,
+                  (err, result) => {
+                    if (err) {
+                      console.log(`apple script error deleting multi login items: ${err}, ${result}`)
+                    }
+                  }
+                )
+              }
+            }
           }
-        })
-      } catch (e) {
-        console.log('Error setting apple startup prefs: ', e)
+        }
+      )
+    } catch (e) {
+      console.log('Error setting apple startup prefs: ', e)
+    }
+  }
+
+  setWinLoginState() {
+    app.setLoginItemSettings({openAtLogin: !!this.state.openAtLogin})
+    // $FlowIssue
```

Comment:
looks like this is because `process.env.whatever` may be null. I think in this case you should either have `process.env.APPDATA || ''` or `if (!process.env.APPDATA) { throw new Error('APPDATA unexpectedly empty') }` or something else instead of `$FlowIssue`